sources/osgUtil/Composer.js

```diff
@@ -355,48 +426,172 @@ define( [
+
+            // copy filter program and uniforms on the 2 cameras
+            st0.setAttributeAndModes( this._program );
+            st1.setAttributeAndModes( this._program );
+
+            if ( this._uniforms ) {
+                for ( var k = 0; k < this._uniforms.length; k++ ) {
+                    st0.addUniform( this._uniforms[ k ] );
+                    st1.addUniform( this._uniforms[ k ] );
+                }
+            }
+
+            st0.addUniform( Uniform.createInt1( 0, 'Texture0' ) );
+            st0.addUniform( Uniform.createInt1( 1, 'Texture1' ) );
+
+            st1.addUniform( Uniform.createInt1( 0, 'Texture0' ) );
+            st1.addUniform( Uniform.createInt1( 1, 'Texture1' ) );
+
+            // copy input on both camera
+            // Composer::Build set the last render into the current filter stateset texture unit 0
+            // we copy that into each camera as Texture unit 0
+            st0.setTextureAttributeAndModes( 0, filterStateSet.getTextureAttribute( 0, 'Texture' ) );
+            st1.setTextureAttributeAndModes( 0, filterStateSet.getTextureAttribute( 0, 'Texture' ) );
+
+            st0.setTextureAttributeAndModes( 1, this._rtt1 );
+            st1.setTextureAttributeAndModes( 1, this._rtt0 );
+
+            // if not the last filter
+            // bind both result to next filter
+            // rtt0 to texture unit 0
+            // rtt0 to texture unit 1
+            if ( i !== array.length - 1 ) {
+
+                // just translate stateset to the next filter
+                var nextSt = array[ i + 1 ].filter.getStateSet();
+
+                nextSt.setTextureAttributeAndModes( 0, this._rtt0 );
+                nextSt.setTextureAttributeAndModes( 1, this._rtt1 );
+
+                nextSt.addUniform( Uniform.createInt1( 0, 'Texture0' ) );
+                nextSt.addUniform( Uniform.createInt1( 1, 'Texture1' ) );
+            }
+
+            var quad = Shape.createTexturedFullScreenFakeQuadGeometry();
+            if ( this.buildGeometry !== undefined )
+                quad = this.buildGeometry( quad );
+            quad.setName( 'composer layer' );
+
+            this._cameraRtt0.addChild( quad );
+            this._cameraRtt1.addChild( quad );
+
+            composer.addChild( this._cameraRtt0 );
+            composer.addChild( this._cameraRtt1 );
+
+            composer._textureRTT.push( this._rtt0 );
+            composer._textureRTT.push( this._rtt1 );
+
+            composer._cameraRTT.push( this._cameraRtt0 );
+            composer._cameraRTT.push( this._cameraRtt1 );
+
+            // hide one of the two pass, as we will render only one each frame
+            this._cameraRtt1.setNodeMask( ~0x0 );
+
+            // last texture result, only one possible so the first will do
+            return this._rtt0;
+        },
+        switch: function () {
+
+            // PingPong
+            if ( this._cameraRtt0.getNodeMask() === ~0x0 ) {
```

Comment:
Totally better code-wise, but really feel more obsuce for code reader. 
Sure you want to optimize ?